hoomd/BoxResizeUpdater.cc

```diff
@@ -45,105 +49,193 @@ BoxResizeUpdater::~BoxResizeUpdater()
     m_exec_conf->msg->notice(5) << "Destroying BoxResizeUpdater" << endl;
     }
 
-/*! \param scale_particles Set to true to scale particles with the box. Set to false to leave particle positions alone
-    when scaling the box.
-*/
-void BoxResizeUpdater::setParams(bool scale_particles)
+void BoxResizeUpdater::setBox1(BoxDim box)
+    {
+    if (!m_py_box1.is_none())
+        {
+        m_py_box1 = m_py_box1.attr("__class__")
+                            .attr("_from_cpp")(box);
+        }
+    m_box1 = box;
+    }
+
+void BoxResizeUpdater::setBox1Py(pybind11::object box)
     {
-    m_scale_particles = scale_particles;
+    m_box1 = getBoxDimFromPyObject(box);
+    m_py_box1 = box;
     }
 
-/*! Perform the needed calculations to scale the box size
+void BoxResizeUpdater::setBox2(BoxDim box)
+    {
+    if (!m_py_box1.is_none())
+        {
+        m_py_box1 = m_py_box1.attr("__class__")
+                            .attr("_from_cpp")(box);
+        }
+    m_box2 = box;
+    }
+
+void BoxResizeUpdater::setBox2Py(pybind11::object box)
+    {
+    m_box2 = getBoxDimFromPyObject(box);
+    m_py_box2 = box;
+    }
+
+/// Get the current box based on the timestep
+BoxDim BoxResizeUpdater::getCurrentBox(unsigned int timestep)
+    {
+    Scalar min = m_variant->min();
+    Scalar max = m_variant->max();
+    Scalar cur_value = (*m_variant)(timestep);
+    Scalar scale = 0;
+    if (cur_value == max)
+        {
+        scale = 1;
+        }
+    else if (cur_value > min)
+        {
+        scale = (cur_value - min) / (max - min);
+        }
+
+    Scalar3 L1 = m_box1.getL();
```

Comment:
`m_py_box` is mostly to just ensure that we can return the box from C++ when asked. These keeps the boxes as values in a `ParameterDict`. I could just make them properties; I honestly wasn't sure what the best solution was, but given the possibility in the future to strive for a full serialization of a HOOMD simulation script, I hedged on the side of use the existing abstractions.

Reply:
Understood. Is there a requirement that the reference stay the same:
```
updater.box1 = b
assert(b is updater.box1)
```
Or is equality all that we need?
```
updater.box1 = b
assert(b == updater.box1)
```
For the latter, we could avoid storing the `m_py_box?` and construct them only when needed in the getter.

Reply:
It depends if a user does this
````python
box_up = BoxResizeUpdater(box1, box2, ...)
box1.scale(1.1)
assert box1 == box_up.box1
````
Should it raise or not. I am inclined to think that this should "work" and not raise, but that is referential (equality would not be enough).

Alternatively, I could do a check if `_cpp_obj` exists for any given object in a `ParameterDict` or `TypeParameter` that only the C++ object gets passed down or up. The problem here is that boxes are currently created so that they use `std::unique_ptr`, so they would have to be changed to use `std::shared_ptr` like the rest.

Reply:
Yes, to enable modifying values by reference you do need to store the reference to the python objects.

Reply:
I guess what I find confusing is that there are two sources of truth: `m_py_box1` and `m_box1`. Even though technically `m_box1` is a shared pointer to an object "owned" by `m_py_box1`. We could simplify by only storing the python boxes, but then we have to have a python/c++ context switch every time we need trigger.

Reply:
Given the current status of `BoxDim` using an `std::unique_ptr` we will have to opt for the context switching to get the references right. I will make a `TODO` in the code to address future changes to make this not require a switch.